exchanges/gateio/gateio_websocket_futures.go

```diff
@@ -672,23 +672,28 @@ func (g *Gateio) processBalancePushData(data []byte, assetType asset.Item) error
 	if err != nil {
 		return err
 	}
-	accountChange := make([]account.Change, len(resp.Result))
-	for x := range resp.Result {
-		info := strings.Split(resp.Result[x].Text, currency.UnderscoreDelimiter)
+	creds, err := g.GetCredentials(ctx)
+	if err != nil {
+		return err
+	}
+	changes := make([]account.Change, len(resp.Result))
+	for x, bal := range resp.Result {
+		info := strings.Split(bal.Text, currency.UnderscoreDelimiter)
 		if len(info) != 2 {
 			return errors.New("malformed text")
 		}
-		code := currency.NewCode(info[0])
-		accountChange[x] = account.Change{
-			Exchange: g.Name,
-			Currency: code,
-			Asset:    assetType,
-			Amount:   resp.Result[x].Balance,
-			Account:  resp.Result[x].User,
+		changes[x] = account.Change{
+			AssetType: assetType,
+			Account:   bal.User,
+			Balance: &account.Balance{
+				Currency:  currency.NewCode(info[0]),
+				Total:     bal.Balance,
+				Free:      bal.Balance,
+				UpdatedAt: bal.Time.Time(),
+			},
 		}
 	}
-	g.Websocket.DataHandler <- accountChange
-	return nil
+	return account.ProcessChange(g.Name, changes, creds)
```

Comment:
I am going to request this and all other instances include `g.Websocket.DataHandler <- changes`

The reason for this is that GoCryptoTrader supports custom websocket DataHandlers and this change prevents users having their own DataHandler from receiving `account.Change` information.

Reply:
OK,  let's keep `g.Websocket.DataHandler <- changes`